c/src/ml-api-remote-service.c

```diff
@@ -218,13 +218,26 @@ _mlrs_model_register (gchar * service_key, nns_edge_data_h data_h,
   GError *error = NULL;
   gboolean ret = TRUE;
 
-  nns_edge_data_get_info (data_h, "description", &description);
-  nns_edge_data_get_info (data_h, "name", &name);
-  nns_edge_data_get_info (data_h, "activate", &activate);
+  if (NNS_EDGE_ERROR_NONE != nns_edge_data_get_info (data_h, "description",
+          &description)
+      || NNS_EDGE_ERROR_NONE != nns_edge_data_get_info (data_h, "name", &name)
+      || NNS_EDGE_ERROR_NONE != nns_edge_data_get_info (data_h, "activate",
+          &activate)) {
+    _ml_loge ("Failed to get info from data handle.");
+    ret = FALSE;
+    goto error;
+  }
+
   active_bool = _mlrs_parse_activate (activate);
 
   dir_path = g_build_path ("/", current_dir, service_key, NULL);
-  g_mkdir_with_parents (dir_path, 0755);
+  if (g_mkdir_with_parents (dir_path, 0755) < 0) {
```

Comment:
Before making a new directory, is it necessary to check that the target directory exists?

Reply:
If the directory already exists, this returns 0. So I think don't need to check the the existence of the directory.